oremda/operator.py

```diff
@@ -53,48 +56,65 @@ def operate(self, info):
             _data_outputs[key] = object_id.binary().hex()
 
         with self.client.open_queue(self.output_queue_name) as output_queue:
-            info = {
+            result = ResultTaskMessage(**{
                 'meta_outputs': meta_outputs,
                 'data_outputs': _data_outputs
-            }
+            })
 
-            output_queue.send(json.dumps(info))
+            output_queue.send(json.dumps(result.dict()))
 
-    def execute(self, meta_inputs, data_inputs_id, params):
-        data_inputs = {}
+    def execute(
+        self,
+        meta_inputs: Dict[PortKey, MetaType],
+        data_inputs_id: Dict[PortKey, ObjectId],
+        params: JSONType
+    ) -> Tuple[Dict[PortKey, MetaType], Dict[PortKey, plasma.ObjectID]]:
+        data_inputs: Dict[PortKey, DataType] = {}
         for key, object_id in data_inputs_id.items():
             data_inputs[key] = self.client.get_object(object_id)
 
         meta_outputs, data_outputs = self.kernel(meta_inputs, data_inputs, params)
 
-        data_outputs_id = {}
+        data_outputs_id: Dict[PortKey, ObjectId] = {}
         for key, array in data_outputs.items():
             data_outputs_id[key] = self.client.create_object(array)
 
         return meta_outputs, data_outputs_id
 
     @abstractmethod
-    def kernel(self, meta, data, parameters):
+    def kernel(
+        self,
+        meta: Dict[PortKey, MetaType],
+        data: Dict[PortKey, DataType],
+        parameters: JSONType
+    ) -> Tuple[Dict[PortKey, MetaType], Dict[PortKey, DataType]]:
         pass
 
+KernelFn = Callable[
+    [Dict[PortKey, MetaType], Dict[PortKey, DataType], JSONType],
+    Tuple[Dict[PortKey, MetaType], Dict[PortKey, DataType]]
+]
 
-def operator(func=None, name=None, start=True,
-             plasma_socket_path=DEFAULT_PLASMA_SOCKET_PATH):
+def operator(func: Optional[KernelFn] = None, _name : Optional[str] = None, start: bool = True,
+             plasma_socket_path: str = DEFAULT_PLASMA_SOCKET_PATH):
 
     # A decorator to automatically make an Operator where the function
     # that is decorated will be the kernel function.
 
-    def decorator(func):
-        nonlocal name
-        if name is None:
+    def decorator(func: KernelFn) -> Operator:
+        nonlocal _name
+
+        if _name is None:
             name = func.__name__
+        else:
+            name = _name
 
         @wraps(func)
-        def kernel(self, *args, **kwargs):
+        def kernel(_, *args, **kwargs):
             # Remove self so the caller does not need to add it
             return func(*args, **kwargs)
 
-        class_name = f'{name.capitalize()}Operator'
+        class_name = f'{name.capitalize()}Operator'      
```

Comment:
Can you set up your text editor to automatically remove trailing whitespace when you save? 

[Here is an example for visual studio code](https://stackoverflow.com/a/53663494).